pages/studentAssessmentInstanceHomework/studentAssessmentInstanceHomework.js

```diff
@@ -101,54 +107,54 @@ router.get('/', function (req, res, next) {
       }
     }
     res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
-  });
-});
-
-router.post('/', function (req, res, next) {
-  if (res.locals.assessment.type !== 'Homework') return next();
-  if (!res.locals.authz_result.authorized_edit) {
-    return next(error.make(403, 'Not authorized', res.locals));
-  }
-
-  if (req.body.__action === 'attach_file') {
-    util.callbackify(studentAssessmentInstance.processFileUpload)(req, res, function (err) {
-      if (ERR(err, next)) return;
-      res.redirect(req.originalUrl);
-    });
-  } else if (req.body.__action === 'attach_text') {
-    util.callbackify(studentAssessmentInstance.processTextUpload)(req, res, function (err) {
-      if (ERR(err, next)) return;
-      res.redirect(req.originalUrl);
-    });
-  } else if (req.body.__action === 'delete_file') {
-    util.callbackify(studentAssessmentInstance.processDeleteFile)(req, res, function (err) {
-      if (ERR(err, next)) return;
-      res.redirect(req.originalUrl);
-    });
-  } else if (req.body.__action === 'leave_group') {
-    if (!res.locals.authz_result.active) return next(error.make(400, 'Unauthorized request.'));
-    groupAssessmentHelper.leaveGroup(
-      res.locals.assessment.id,
-      res.locals.user.user_id,
-      res.locals.authn_user.user_id,
-      function (err) {
+  })
+);
+
+router.post(
+  '/',
+  asyncHandler(async function (req, res, next) {
+    if (res.locals.assessment.type !== 'Homework') return next();
+    if (!res.locals.authz_result.authorized_edit) {
+      return next(error.make(403, 'Not authorized', res.locals));
```

Comment:
```suggestion
      throw error.make(403, 'Not authorized', res.locals);
```

Ditto elsewhere in the function.